src/BlockTypes/AbstractBlock.php

```diff
@@ -181,14 +188,26 @@ protected function register_block_type() {
 			'editor_script'   => $this->get_block_type_editor_script( 'handle' ),
 			'editor_style'    => $this->get_block_type_editor_style(),
 			'style'           => $this->get_block_type_style(),
-			'attributes'      => $this->get_block_type_attributes(),
-			'supports'        => $this->get_block_type_supports(),
 		];
 
 		if ( isset( $this->api_version ) && '2' === $this->api_version ) {
 			$block_settings['api_version'] = 2;
 		}
 
+		// Prefer to register with metadata if the path is set in the block's class.
+		if ( isset( $this->metadata_path ) && '' !== $this->metadata_path ) {
+			register_block_type(
+				$this->asset_api->get_block_metadata_path( $this->metadata_path ),
+				$block_settings
+			);
+			return;
+		}
+
+		// Insert attributes and supports if we're not registering the block using metadata.
+		// These are added after because registering with metadata caused attributes and supports to be empty arrays.
```

Comment:
I don't understand this comment - we're registering with metadata further up and returning, so why would it affect this?

Reply:
Oops, I made a typo, the comment makes more sense now I've reworded it.

Essentially, if `$block_settings['supports']` or `$block_settings['attributes']` is set, then it overrides what is in metadata so we don't want to pass them to `register_block_type` if we're also using metadata.